### PR TITLE
doc: rules for developing (with) `datalad-core`

### PR DESCRIPTION
**Pull request by mih**
`hatch env show` is mentioned in the rules. Here is what this shows, in order to illustrate contributors what functionality is available locally:

image

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+Tests should be silent on stdout/stderr as much as possible.
+In particular (but not only), result renderings of DataLad commands must no be produced, unless necessary for testing a particular feature.
+
+
+## CI workflows
+
+The addition of automation via CI workflows is welcome.
+However, such workflows should not force developers to depend on, or have to wait for any particular service to run a workflow before they can discover essential outcomes.
+When such workflows are added to online services, an equivalent setup for local execution should be added to the repository.
+The `hatch` environments and tailored commands offer a straightforward, and discoverable method to fulfill this requirement (`hatch env show`).
```
Comment (yarikoptic) [inline CONTRIBUTING.md:91]: Should have at least minimal section on `cz` which seems to be used to 
- enforce commit messages consistency
- releases? (ref #7)

Comment (mih) [reply CONTRIBUTING.md:189]: A full developer cheat sheet was added.

**Comment by mih** [comment]
https://github.com/datalad/datasalad is now aligned with this setup/rules. https://github.com/datalad/datalad-next will be after https://github.com/datalad/datalad-next/pull/752

**Comment by mih** [comment]
I will leave this open till tomorrow and then merge, if there is no more feedback.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+Contributors should strive for files with less than 500 lines of code.
+
+Within a sub-package, tests should import the tested code via relative imports.
+
+Code users should be able to import the most relevant functionality from the sub-package's `__init__.py`.
+Only items importable from the sub-package's top-level are considered to be part of its "public" API.
+If a sub-module is imported in the sub-package's `__init__.py`, consider adding `__all__` to the sub-module to restrict wildcard imports from the sub-module, and to document what is considered to be part of the "public" API.
+
+Sub-packages should be as self-contained as possible.
+When functionality is shared between sub-packages, absolute imports should be made.
```
Comment (christian-monch) [inline CONTRIBUTING.md:169]:
```suggestion
If functionality is shared between sub-packages, absolute imports should be made.
```

**Review by christian-monch — APPROVED** [review]
Nice (only a small comment from me)

**Pull request merged**